presto-orc/src/main/java/com/facebook/presto/orc/writer/StreamOrderingLayout.java

```diff
@@ -141,6 +141,12 @@ private StreamMetadata getStreamMetadata(
         // iterate through all the encodings and if the encoding has additional sequence encodings put it in the map
         for (Map.Entry<Integer, ColumnEncoding> entry : nodeIdToColumnEncodings.entrySet()) {
             int nodeId = entry.getKey();
+
+            // skip the root node, because it doesn't have a column
+            if (nodeId == 0) {
```

Comment:
Could it fail if nodeId does not have any columns ? This can happen, say you are encoding List<Integer> and all the List elements are null. In which case Integer will not have any representations at all. 

So would be easier to check if the nodeIdToColumn has no nodeId and continue.

Reply:
No, this should not happen because nodeIdToColumn is populated from the schema (I also verified this).
We are seeing this issue because column encoding for the root node 0 is force-added to the column encodings by the orc writer.

Reply:
I added a test case for empty complex type just in case to make sure we are not hitting NPE.